lib/apimatic-core/types/error_case.rb

```diff
@@ -68,7 +68,12 @@ def _get_resolved_error_message_template(response)
                                                                        error_message_template)
 
       # Handling response body placeholder
-      response_payload = ApiHelper.json_deserialize(response.raw_body, true)
+      begin
+        response_payload = ApiHelper.json_deserialize(response.raw_body, true) unless response.raw_body.nil?
+      rescue TypeError
+        response_payload = nil
```

Comment:
Question: Is this unit tested? If not, kindly add a unit test against this case.

Reply:
Test case added.